File: Replace_Exception_with_Test.py
# After
def getValueForPeriod(periodNumber):
    try:
        return values[periodNumber]
    except IndexError:
        return 0

# Refactoring
def getValueForPeriod(self, periodNumber):
    if periodNumber >= len(self.values):
        return 0
    return self.values[periodNumber]

File: Replace_Exception_with_Test.js
//After
function getValueForPeriod(periodNumber)  {
  try {
    return values[periodNumber];
  } catch(IndexError) {
    return 0;
  }    
}        

// Refactoring
function getValueForPeriod(self, periodNumber) {
  if (periodNumber >= self.values.length) {
    return 0;
  }    
  return self.values[periodNumber];
}    
